### DISCUSSION TITLE
SFTP Storage not work

### DISCUSSION THREAD

**Discussion by Rainshaw** [requarks/wiki#3538] [Error / Bug Report] [answered]
**Describe the bug**
Due to the clain

> Base directory where files will be transferred to. The path must already exists and be writable by the user.

I found that not only base dir need to be create manually but also the sub folders do need!
if there is some dir not exists then will come to 

> info: (STORAGE/SFTP) Adding page aaa/bbb/ccc.md...
> warn: No such file

**To Reproduce**
1. Go to setting and set up SFTP 
2. Click on RUN
3. Scroll down to the docker log

**Expected behavior**
subfolder auto create

**Screenshots**

**Host Info (please complete the following information):**
 - OS: Docker
 - Wiki.js version: 2.4.107
 - Database engine: postgres 11.8

**Additional context**

**Comment by dualmacops** [comment]
Same problem.

Host Info:

OS: Docker
Wiki.js version: 2.5.170
Database engine: postgres 11.9

**Comment by dualmacops** [comment]
Well the problem is not exactly the same, when i activate the sftp i get:
info: (STORAGE/SFTP) Initializing...
warn: Cannot read property 'message' of null

And when i click export all i get:
info: (STORAGE/SFTP) Exporting all content to the remote server...
info: (STORAGE/SFTP) Adding page home.html...
warn: Cannot read property 'writeFile' of null

**Comment by Jamesits** [comment]
Can reproduce. If a new folder is created, SFTP storage backend won't create the corresponding directory. 
Environment:
* Wiki.js 2.5.170 (Official Docker image)
* Database: postgres
* SFTP backup destination: TrueNAS 12.0 (destination permission set to 700)

**Comment by dualmacops** [comment]
For now i fixed my problem by doing a local storage and using RCLONE to move the files to the sftp.
Also i had to use the --sftp-disable-hashcheck in rclone to get it to work, so it might me a SFTP server fault in my case.

**Comment by Rainshaw** [comment, marked as answer]
fix by #4024
